File: PizzaBackendServer/WebApi/BLL/Models/OrderModel.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BLL.Interfaces;
using DAL.Data;

namespace BLL.Models
{
    public class OrderModel
    {
        public int Id { get; set; }
        public DateTime CreationTime { get; set; }
        public DateTime? FinishedTime { get; set; }
        public DateTime? PredictedTime { get; set; }
        public decimal Price { get; set; }
        public string Address { get; set; }
        public string PhoneNumber { get; set; }
        public string ClientName { get; set; }
        public string PayingType { get; set; }
        public int? UsedBonuses { get; set; }
        public int GivenBonuses { get; set; }
        public string? Comment { get; set; }
        public decimal? Change { get; set; }
        public int ClientId { get; set; }
        public int? WorkerId { get; set; }
        public string? WorkerName { get; set; }
        public int? CourierId { get; set; }
        public string? CourierName { get; set; }
        public string? StatusName { get; set; }

        public int StatusId { get; set; }
        public List<OrderStringModel> OrderLines { get; set; }
        public List<int> OrderStringIds { get; set; }
      //  public StatusModel Status { get; set; }
        public OrderModel() { }

        public OrderModel(Order o, IUnitOfWork dataBase)
        {
            Id = o.Id;
            CreationTime = o.CreationTime;
            FinishedTime = o.FinishedTime;
            PredictedTime = o.PredictedTime;
            Price = o.Price;
            Address = o.Address;
            PhoneNumber = o.PhoneNumber;
            ClientName = o.ClientName;
            PayingType = o.PayingType;
            Change = o.Change;
            ClientId = o.ClientId;
            WorkerId = o.WorkerId;
            CourierId = o.CourierId;
            StatusId = o.StatusId;
            UsedBonuses = o.UsedBonuses;
            GivenBonuses = o.GivenBonuses;
            Comment = o.Comment;
            var Status = dataBase.StatusRepository.Get(o.StatusId);
            if (Status != null) {
                StatusName = Status.Name;
            }
            //OrderStringIds = o.OrderStrings.Select(i => i.Id).ToList();
            OrderLines = dataBase.OrderStringRepository.GetAll().Select(i => new OrderStringModel(i, dataBase)).Where(i => i.OrderId == Id).ToList();
            if (WorkerId != null) {
                var worker = dataBase.UserRepository.Get((int)WorkerId);
                WorkerName = worker.Name;
            }
            if (CourierId != null)
            {
                var courier = dataBase.UserRepository.Get((int)CourierId);
                CourierName = courier.Name;
            }
            //foreach (var os in o.OrderStrings)
            //{
            //    OrderLines.Add(new OrderStringModel(os));
            //}
        }
    }
}

File: PizzaMobileApp/pizzer_mobile/lib/models/order_model.dart
import 'dart:io';
import 'package:pizzer_mobile/models/order_lines_model.dart';

class OrderModel {
  int? id;
  DateTime? creationTime;
  DateTime? finishedTime;
  DateTime? predictedTime;
  double? price;
  String? address;
  String? phoneNumber;
  String? clientName;
  String? payingType;
  int? usedBonuses;
  int? givenBonuses;
  String? comment;
  double? change;
  int? clientId;
  int? workerId;
  String? workerName;
  int? courierId;
  String? courierName;
  String? statusName;
  int? statusId;
  List? orderLines;

  OrderModel({
    this.id,
    this.creationTime,
    this.finishedTime,
    this.predictedTime,
    this.price,
    this.address,
    this.phoneNumber,
    this.clientName,
    this.payingType,
    this.usedBonuses,
    this.givenBonuses,
    this.comment,
    this.change,
    this.clientId,
    this.workerId,
    this.workerName,
    this.courierId,
    this.courierName,
    this.statusName,
    this.statusId,
    this.orderLines,
  });

  OrderModel.fromJsonSingle(dynamic json) {
    id = json['id'];
    creationTime = DateTime.tryParse(json['creationTime']);
    if (json['finishedTime'] != null) {
      finishedTime = DateTime.tryParse(json['finishedTime']);
    }
    predictedTime = DateTime.tryParse(json['predictedTime']);
    price = json['price'];
    address = json['address'];
    phoneNumber = json['phoneNumber'];
    clientName = json['clientName'];
    payingType = json['payingType'];
    if (json['usedBonuses'] != null) {
      usedBonuses = json['usedBonuses'];
    }
    givenBonuses = json['givenBonuses'];
    if (json['comment'] != null) {
      comment = json['comment'];
    }
    change = json['change'];
    clientId = json['clientId'];
    workerId = json['workerId'];
    workerName = json['workerName'];
    courierId = json['courierId'];
    courierName = json['courierName'];
    statusName = json['statusName'];
    statusId = json['statusId'];
    orderLines =
        json['orderLines'].map((e) => OrderLinesModel.fromJson(e)).toList();
  }

  // OrderModel.fromJson(dynamic json) {
  //   id = json['id'];
  //   email = json['email'];
  //   name = json['name'];
  //   role = json['role'];
  //   isApproved = json['isApproved'];
  //   phone = json['phone'];
  //   bonuses = json['bonuses'];
  //   passport = json['passport'];
  // }
}
